internal/scheduler/metrics/metrics.go

```diff
@@ -59,8 +59,19 @@ type Metrics struct {
 	matchedRegexIndexByErrorMessage *lru.Cache
 
 	// Job metrics.
-	transitions     *prometheus.CounterVec
-	resourceSeconds *prometheus.CounterVec
+	jobs             *prometheus.CounterVec
```

Comment:
Make this a map instead. We can't know in advance which resources exist within the system as nodes may export arbitrary resources.